### PR TITLE
Add minio as a system service (#185)

### PR DESCRIPTION
**Pull request by fredrikhgrelland**

### CODE REVIEW

File: ansible/install.yml [hunk 1]
```diff
@@ -18,6 +18,15 @@
...
+    - name: Download minio
+      get_url:
+        url: https://dl.min.io/server/minio/release/linux-amd64/minio
+        dest: /usr/local/bin/minio
+        mode: 'a+x'
+
+    - name: Install systemd unit file minio
+      template: src=minio.service.j2 dest=/etc/systemd/system/minio.service
+
     - name: Install systemd unit file
```
Comment (pdmthorsrud) [inline ansible/install.yml:30]:
```suggestion
    - name: Install systemd unit files hashistack
```
Might as well change this while we're at it.

File: README.md [hunk 2]
```diff
@@ -58,9 +58,11 @@ This repo will build a base-box for different projects to extend on. The base bo
 
 The default box will start Nomad, Vault and Consul, bound on loopback and advertising on the ip `10.0.3.10`, which should be available on your local machine.
 Portforwarding for nomad on port `4646` should bind to `127.0.0.1` and should allow you to use the nomad binary to post jobs directly. Consul and Vault has also been portforwarded, and are also available on `127.0.0.1` on port `8500` and `8200` respectively.
+Minio is started on port `9000` and sharing /vagrant (your repo) from within the vagrant box. This is used to present build artifacts to nomad over s3.
```
Comment (pdmthorsrud) [inline README.md:61]:
```suggestion
Minio is started on port `9000` and sharing /vagrant (your repo) from within the vagrant box.
```
There could very quickly be other uses for this, so I don't feel like that sentence is necessary.

File: README.md [hunk 3]
```diff
@@ -58,9 +58,11 @@ This repo will build a base-box for different projects to extend on. The base bo
 
 The default box will start Nomad, Vault and Consul, bound on loopback and advertising on the ip `10.0.3.10`, which should be available on your local machine.
```
Comment (pdmthorsrud) [inline README.md:59]:
```suggestion
The default box will start Nomad, Vault, Consul, and MinIO bound on loopback and advertising on the ip `10.0.3.10`, which should be available on your local machine.
```

**Comment by pdmthorsrud** [comment]
What about portforwarding for the MinIO instance?

**Comment by fredrikhgrelland** [comment]
> What about portforwarding for the MinIO instance?

Not necessary, but can easily be added in the future.

**Comment by pdmthorsrud** [comment]
> > What about portforwarding for the MinIO instance?
> 
> Not necessary, but can easily be added in the future.

Ok, I guess you want this in as quick as possible, so fair enough.

**Comment by fredrikhgrelland** [comment]
> > > What about portforwarding for the MinIO instance?
> > 
> > 
> > Not necessary, but can easily be added in the future.
> 
> Ok, I guess you want this in as quick as possible, so fair enough. It does break the pattern, and doesn't make much sense to **not** have considering the other services being available, but we'll make an issue for that then!

FYI. It is available, just not on 127.0.0.1:9000 on the host machine. The reason why we have this for vault/consul/nomad is that they will by default work with localhost, and such are we able to use the cli directly. This is not the case with minio, as it is a network system and the cli is not currently added

**Comment by zhenik** [comment]
@fredrikhgrelland So port 9000 is taken, If we want to run any nomad jobs with minio (no consul-connect), we need to allocate another port, correct?

**Comment by fredrikhgrelland** [comment]
@zhenik correct. But we should always use the dynamic ports for any service :)

**Pull request merged**